lib/eblox/data/taxonomy.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Eblox.Data.Taxonomy do
+  @moduledoc false
+
+  use Supervisor
+
+  @type post_id() :: term()
+  @type registry_option() :: Registry.start_option()
+
+  @callback registry_options([registry_option()]) :: [registry_option()]
+  @callback on_add(module(), post_id(), term()) :: :ok | :error
+  @callback on_remove(module(), post_id()) :: :ok
+
+  defmodule Meta do
+    @moduledoc false
+
+    use Agent
+
+    alias Eblox.Data.Taxonomy.Meta
+
+    @type t :: %{
+      __struct__: Meta,
+      impl: module(),
+    }
+    @enforce_keys ~w|impl|a
+    defstruct ~w|impl|a
+
+    def start_link([{:impl, impl} | opts]) do
+      Agent.start_link(fn -> %Meta{impl: impl} end, opts)
```

Comment:
This literally enforces the caller to care about the order of elements in the `opts` list which is not expected. There are two ways of doing this in a more idiomatic manner:

```elixir
def start_link(impl, opts) do
```

and/or

```suggestion
    def start_link(opts) do
      {impl, opts} = Keyword.pop!(opts, :impl) # we’re to raise if no `impl` given
      Agent.start_link(fn -> %Meta{impl: impl} end, opts)
```

I personally prefer the latter because it makes it easier to scale (when the second, third, fourth param would be needed to initialize `Meta`.) Also, `Keyword.split/2` might be of some help.